launcher/ui/pages/modplatform/ModModel.cpp

```diff
@@ -6,12 +6,23 @@
 
 #include "minecraft/MinecraftInstance.h"
 #include "minecraft/PackProfile.h"
+#include "minecraft/mod/ModFolderModel.h"
+#include "modplatform/ModIndex.h"
 
 #include <QMessageBox>
 
 namespace ResourceDownload {
 
-ModModel::ModModel(BaseInstance const& base_inst, ResourceAPI* api) : ResourceModel(api), m_base_instance(base_inst) {}
+ModModel::ModModel(BaseInstance const& base_inst, ResourceAPI* api) : ResourceModel(api), m_base_instance(base_inst)
+{
+    auto folder = static_cast<MinecraftInstance const&>(m_base_instance).loaderModList();
+    for (auto mod : folder->allMods()) {
+        auto meta = mod->metadata();
+        ModPlatform::IndexedPack pack{ meta->project_id, meta->provider, meta->name, meta->slug };
```

Comment:
not all mods have metadata, in which cases `meta` is `nullptr`, and this line segfaults.

Reply:
I will move this to other PR( to make sure I keep this one clean)